source/part1/chapter-02/md/01-data-manipulation.md

```diff
@@ -59,52 +51,49 @@ data
 <!-- #endregion -->
 
 ```python deletable=true editable=true jupyter={"outputs_hidden": false}
-#Calculate max min difference
+# Calculate max min difference
 data['DIFF'] = data['MAX'] - data['MIN']
 
 # Check the result
 data.head()
 ```
 
 <!-- #region deletable=true editable=true -->
-The calculations were stored into the ``DIFF`` column as planned. 
+The calculations were stored into the ``DIFF`` column as planned. Conducting calculation like this is extremely fast in pandas because the math operations happen in *vectorized* manner. This means that instead of looping over individual values of the DataFrame and comparing them to each other, calculating the difference happens simultaneously at all rows.
 
-You can also create new columns on-the-fly at the same time when doing the calculation (the column does not have to exist before). Furthermore, it is possible to use any kind of math
-algebra (e.g. subtracttion, addition, multiplication, division, exponentiation, etc.) when creating new columns.
-<!-- #endregion -->
+You can also create new columns on-the-fly when doing the calculation (i.e. the column does not have to exist before). Furthermore, it is possible to use any kind of math algebra (e.g. subtracttion, addition, multiplication, division, exponentiation, etc.) when creating new columns.
 
-<!-- #region deletable=true editable=true -->
 We can for example convert the Fahrenheit temperatures in the `TEMP` column into Celsius using the formula that we have seen already many times. Let's do that and store it in a new column called `TEMP_CELSIUS`.
 <!-- #endregion -->
 
 ```python deletable=true editable=true jupyter={"outputs_hidden": false}
 # Create a new column and convert temp fahrenheit to celsius:
 data['TEMP_CELSIUS'] = (data['TEMP'] - 32) / (9/5)
 
-#Check output
+# Check output
 data.head()
 ```
 
-#### Check your understanding
+**Check your understanding**
 
-Calculate the temperatures in Kelvins using the Celsius values **and store the result a new column** calle `TEMP_KELVIN` in our dataframe.
-    
-0 Kelvins is is -273.15 degrees Celsius as we learned during [Lesson 4](https://geo-python-site.readthedocs.io/en/latest/notebooks/L4/functions.html#let-s-make-another-function).
+**NOTE: THIS IS QUITE COMPLICATED TASK. MAKE AN EASIER ONE THAT DOES NOT REQUIRED GOING BACK TO OTHER MATERIALS.** Calculate the temperatures in Kelvins using the Celsius values **and store the result a new column** calle `TEMP_KELVIN` in our dataframe.
```

Comment:
If we provide the formula for Kelvin calculation, shouldn't be too difficult? The solution is data["TEMP_KELVIN"] = data["TEMP_CELSIUS"] + 273.15 

So we could add that "Here is the formula for converting Celsius degrees (C) to Kelvins (K): K = C + 273.15"

Reply:
Ah that's correct. 🙂 I quicky checked that this would require using the function stuff from the Chapter 1 (mixed with the fahrenheit things or something). But indeed that is a simple conversion so let's keep it as it is with the formula. 👍🏻

Reply:
yes. haha, I guess the idea with that instruction was to make the students think a bit, but the solution is eventually super easy :D